bundles/communities-ugc-migration/src/main/java/com/adobe/communities/ugc/migration/importer/ScoresSimpleImportServlet.java

```diff
@@ -0,0 +1,123 @@
+
+
+    @Reference(cardinality = ReferenceCardinality.MANDATORY_UNARY, policy = ReferencePolicy.STATIC)
+    private UserPropertiesService userPropertiesService;
+
+    @Reference
+    private ScoringService scoringService;
+
+
+    /**
+    *The post operation accepts a json file, parses it and applies the profile scores to local profiles
+    * @param request - the request
+    * @param response - the response
+    * @throws javax.servlet.ServletException
+    * @throws java.io.IOException
+    */
+    @Override
+    protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response)
+            throws ServletException, IOException {
+
+        ResourceResolver resolver = request.getResourceResolver();
+        UGCImportHelper.checkUserPrivileges(resolver, resourceResolver);
+        final RequestParameter[] fileRequestParameters = request.getRequestParameters("file");
+
+        String resourcePath = "";
+        if(!request.getParameter("path").equals(null) && !request.getParameter("path").equals("")){
+            resourcePath = request.getParameter("path");
+        } else {
+            log.error("Invalid communities-page JCR content path");
+            System.exit(0);
+        }
+
+        if (fileRequestParameters != null && fileRequestParameters.length > 0 && !fileRequestParameters[0].isFormField()
+            && fileRequestParameters[0].getFileName().endsWith(".json")) {
+                
+            final InputStream inputStream = fileRequestParameters[0].getInputStream();
+            final JsonParser jsonParser = new JsonFactory().createParser(inputStream);
+            JsonToken token = jsonParser.nextToken();
+            
+            if (token.equals(JsonToken.START_OBJECT)) {
+                try {
+                    final UserPropertiesManager userManager = userPropertiesService.createUserPropertiesManager(resolver);
+                    readJsonUserScore(jsonParser, resolver, resourcePath);
+                } catch (RepositoryException e) {
+                    throw new ServletException("Unable to communicate with Jcr repository", e);
+                }
+            } else {
+                throw new ServletException("Expected a start object token, got " + token);
+            }
+        } else {
+            log.error("Invalid file");
+        }
+    }
+
+    private void readJsonUserScore(final JsonParser jsonParser, ResourceResolver resolver, String resourcePath)
+            throws ServletException, IOException {
+
+        JsonToken jsonToken = jsonParser.nextToken();
+
+        Resource componentResource = resolver.getResource(resourcePath);
+        Resource scoreRuleResource = resolver.getResource(resourcePath + "/jcr:content");
+        ValueMap valueMap = scoreRuleResource.adaptTo(ValueMap.class);
+        String scoringProperty = valueMap.get("scoringRules", "");
+
+        if (!scoringProperty.equals(null) && !scoringProperty.equals("")) {
```

Comment:
To elaborate on my earlier comment, I meant adding null checks for the `Resource` objects since `getResource()` can return null. Not sure if it's necessary to check for the scoring property, but probably doesn't hurt.